src/rpc/blockchain.cpp

```diff
@@ -387,23 +398,31 @@ static RPCHelpMan waitforblockheight()
     NodeContext& node = EnsureAnyNodeContext(request.context);
     Mining& miner = EnsureMining(node);
 
-    auto block{CHECK_NONFATAL(miner.getTip()).value()};
     const auto deadline{std::chrono::steady_clock::now() + 1ms * timeout};
 
-    while (IsRPCRunning() && block.height < height) {
+    std::optional<BlockRef> block{miner.getTip()};
+    // If getTip() returns null, which can happen during startup, pass 0 to
+    // waitTipChanged() so it will wait for any tip hash to be set.
+    uint256 tip_hash{block ? block->hash : uint256{}};
+    int block_height{block ? block->height : 0};
+
+    while (block_height < height) {
```

Comment:
Same here, if the user provides `height=0`, and `getTip()` returns `nullopt`, then this will be `while (0 < 0)`, will never enter and `block` will remain `nullopt` and be dereferenced at the end.